core/websock.js

```diff
@@ -107,36 +107,27 @@ export default class Websock {
     }
 
     rQshiftBytes(len) {
-        if (typeof(len) === 'undefined') { len = this.rQlen(); }
+        if (typeof(len) === 'undefined') { len = this.rQlen; }
         this._rQi += len;
         return new Uint8Array(this._rQ.buffer, this._rQi - len, len);
     }
 
     rQshiftTo(target, len) {
-        if (len === undefined) { len = this.rQlen(); }
+        if (len === undefined) { len = this.rQlen; }
         // TODO: make this just use set with views when using a ArrayBuffer to store the rQ
         target.set(new Uint8Array(this._rQ.buffer, this._rQi, len));
         this._rQi += len;
     }
 
-    rQwhole() {
-        return new Uint8Array(this._rQ.buffer, 0, this._rQlen);
-    }
-
-    rQslice(start, end) {
-        if (end) {
-            return new Uint8Array(this._rQ.buffer, this._rQi + start, end - start);
-        } else {
-            return new Uint8Array(this._rQ.buffer, this._rQi + start, this._rQlen - this._rQi - start);
-        }
+    rQslice(start, end = this.rQlen) {
```

Comment:
Default parameters are not supported by IE unfortunately. I guess we did not have a test that caught this?

Reply:
Neither are arrow functions.
I thought that that is why noVNC is transpiled?

Reply:
Sorry, you're right. We just need to confirm that the current transpile settings catch this then.